private/rules/coursier.bzl

```diff
@@ -257,11 +258,15 @@ def _get_aar_import_statement_or_empty_str(repository_ctx):
         return ""
 
 def _java_path(repository_ctx):
-    java_home = repository_ctx.os.environ.get("JAVA_HOME")
-    if java_home != None:
-        return repository_ctx.path(java_home + "/bin/java")
-    elif repository_ctx.which("java") != None:
-        return repository_ctx.which("java")
+    embedded_java = "../bazel_tools/jdk/bin/java"
+    if _is_file(repository_ctx, embedded_java):
+        return repository_ctx.path(embedded_java)
+    else:
+        java_home = repository_ctx.os.environ.get("JAVA_HOME")
+        if java_home != None:
+            return repository_ctx.path(java_home + "/bin/java")
+        elif repository_ctx.which("java") != None:
```

Comment:
I see we did the same thing here, but I don't think it makes things clearer. *shrug*